app/src/components/Courses/CourseComponent.tsx

```diff
@@ -428,17 +437,21 @@ export const CourseComponent = ({ token, id, setTickChange, setId, courseData, u
             </label>
 
             <label
-              htmlFor="course-create"
-              className="whitespace-nowrap h-12 p-2 bg-primary hover:bg-primary focus:ring-blue-500 focus:ring-offset-blue-200 text-white w-full transition ease-in duration-200 text-center text-lg font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg"
-            >
-              <button
-                type="submit"
-                id="addCourse"
-                className="flex items-center justify-center py-4 px-8 h-full w-full cursor-pointer"
-              >
-                Adicionar seções {/** Add sections */}
-              </button>
-            </label>
+          htmlFor="course-create"
+          className="whitespace-nowrap h-12 p-2 bg-primary hover:bg-primary focus:ring-blue-500 focus:ring-offset-blue-200 text-white w-full transition ease-in duration-200 text-center text-lg font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg"
+        >
+          <button
+            type="submit"
+            id="addCourse"
+            disabled={submitLoading}
+            className="flex items-center justify-center py-4 px-8 h-full w-full cursor-pointer"
+          >
+            {submitLoading ? (
+              <span className="spinner-border animate-spin inline-block w-4 h-4 border-2 border-t-transparent rounded-full mr-2"></span>
+            ) : null}
```

Comment:
Maybe this should be its own component so other devs have an easier time using it